### PR TITLE
fix: addresses bug with blank nodes that was breaking nesting

### LINKED ISSUE #91: Nesting and frames support
```
{
  "@context": {"a": "https://example.com/a", "a1": "https://example.com/a1", "b": "https://example.com/b", "c": "https://example.com/c"},
  "a":{
    "b": {
      "c": "hello"
    }
  },
  "a1": "hello2"
}
```

```
{
  "@context": {"a": "https://example.com/a", "a1": "https://example.com/a1", "b": "https://example.com/b", "c": "https://example.com/c"},
 "@explicit": true,
  "a": {
   "@explicit": true,
    "b":{}
   }
}
```

trying to figure out how to get deriveProof on nest claims, for example:

```
{
    "@context": [
      "https://www.w3.org/2018/credentials/v1",
      "https://www.w3.org/2018/credentials/examples/v1",
      "https://w3c-ccg.github.io/ldp-bbs2020/context/v1"
    ],
    "id": "http://example.gov/credentials/3732",
    "type": [
      "VerifiableCredential",
      "UniversityDegreeCredential"
    ],
    "issuer": {
      "id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s"
    },
    "issuanceDate": "2020-03-10T04:24:12.164Z",
    "credentialSubject": {
      "id": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM",
      "degree": {
        "type": "BachelorDegree",
        "name": "Bachelor of Science and Arts"
      }
    },
    "proof": {
      "type": "BbsBlsSignature2020",
      "created": "2020-12-06T16:10:24Z",
      "proofPurpose": "assertionMethod",
      "proofValue": "sKNVGJssWdZcvmCqvcfgz7s64Ibj0ZfqSmZIwgy+yIhCUOlWflsBKCfVNGAPfd8yW/pryt87c7ho/PAJofdFZZB97gpQkpisI/+Rcs2xLtVRrpTiu4YU3sXEFbgPjA3GY/fZOpdFTABVutDvckchLw==",
      "verificationMethod": "did:key:z5TcESXuYUE9aZWYwSdrUEGK1HNQFHyTt4aVpaCTVZcDXQmUheFwfNZmRksaAbBneNm5KyE52SdJeRCN1g6PJmF31GsHWwFiqUDujvasK3wTiDr3vvkYwEJHt7H5RGEKYEp1ErtQtcEBgsgY2DA9JZkHj1J9HZ8MRDTguAhoFtR4aTBQhgnkP4SwVbxDYMEZoF2TMYn3s#zUC7LTa4hWtaE9YKyDsMVGiRNqPMN3s4rjBdB3MFi6PcVWReNfR72y3oGW2NhNcaKNVhMobh7aHp8oZB3qdJCs7RebM2xsodrSm8MmePbN25NTGcpjkJMwKbcWfYDX7eHCJjPGM"
    }
  }
```

with 

```
   const deriveProofFrame = {
      "@context": [
        "https://www.w3.org/2018/credentials/v1",
        "https://www.w3.org/2018/credentials/examples/v1",
        "https://w3c-ccg.github.io/ldp-bbs2020/context/v1",
      {
        "degree": "https://example.com/degree"
      }
      ],
      "type": ["VerifiableCredential", "UniversityDegreeCredential"],
      "@explicit": true,
      "credentialSubject": {
        "@explicit": true,
        degree: {
          name: {}
        }
      }
    }

```

### PR DESCRIPTION
**Pull request by kdenhartog**
## Description

the blank nodes were only being transformed in the node identifier portion of the nQuad. This caused
issues in how the nQuads would be converted back from RDF to JSON which in-inadvertently broke the
ability for a proof to be derived from a nested structure. This fix makes it so that all blank node
identifiers are transformed properly rather than just the node identifiers (first part of the nQuad).

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] The commit message(s) follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] Documentation has been added / updated (for bug fixes / features)
- [X] Changes follow the **[contributing](../docs/CONTRIBUTING.md)** document.

## Motivation and Context

fixes #91

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## Which merge strategy will you use?

- [X] Squash
- [ ] Rebase (REVIEW COMMITS)

### CODE REVIEW

File: __tests__/__fixtures__/data/test_nested_reveal_full_document.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/v1",
+    "https://www.w3id.org/security/v3-unstable"
+  ],
+  "type": ["VerifiableCredential", "UniversityDegreeCredential"],
+  "@exlicit": true
```
Comment (tplooker) [inline __tests__/__fixtures__/data/test_nested_reveal_full_document.json:8]: This is spelt wrong and probably not needed as the default is false when you are intending to reveal a whole frame

File: __tests__/test.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+[
```
Comment (tplooker) [inline __tests__/test.json:1]: Is this needed?

File: src/BbsBlsSignatureProof2020.ts [hunk 3]
```diff
@@ -252,14 +254,12 @@ export class BbsBlsSignatureProof2020 extends suites.LinkedDataProof {
...
+          const prefixIndex = element.indexOf("<urn:bnid:");
+          const closingIndex = element.indexOf(">", prefixIndex);
           return element.replace(
-            nodeIdentifier,
-            nodeIdentifier.substring(
-              "<urn:bnid:".length,
-              nodeIdentifier.length - 1
-            )
+            element.substring(prefixIndex, closingIndex + 1),
+            element.substring(prefixIndex + 10, closingIndex)
```
Comment (tplooker) [inline src/BbsBlsSignatureProof2020.ts:262]: Hmm, why `+ 10` here?

Comment (kdenhartog) [reply src/BbsBlsSignatureProof2020.ts:262]: I'm glad you caught this cause I didn't have a better solution `<urn:bnid:` is 10 characters it's definitely magic string style so I can add some comments in here if we leave this solution

File: __tests__/test.json (hunk 2 above)
Comment (kdenhartog) [reply __tests__/test.json:1]: Nahh good catch, will remove it. This was a temp file I made when I was debugging.

File: src/BbsBlsSignatureProof2020.ts (hunk 3 above)
Comment (tplooker) [reply src/BbsBlsSignatureProof2020.ts:262]:
```suggestion
            element.substring(prefixIndex + "<urn:bnid".length(), closingIndex)
```

**Review by tplooker — APPROVED** [review]

**Pull request merged**